app/routes.py

```diff
@@ -1,2 +1,286 @@
-from flask import Blueprint
+import re
+from flask import Blueprint, request, make_response, jsonify
+from flask_sqlalchemy import model
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from datetime import datetime
+import requests
+import os
+from tests.conftest import completed_task
 
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+
+
+@goals_bp.route("", methods=["GET", "POST"])
+def handle_goals():
+    if request.method == "GET":
+        goals = Goal.query.all()
+        goals_response = []
+        for goal in goals:
+            goals_response.append({
+                "id": goal.goal_id,
+                "title": goal.title
+            })
+
+        return jsonify(goals_response)
+
+    
+    elif request.method == "POST":
+        request_body = request.get_json()
+        if request_body.get("title"):
+            new_goal = Goal(title=request_body["title"])
+            db.session.add(new_goal)
+            db.session.commit()
+            response_value = {"goal":{
+                    "id": new_goal.goal_id,
+                    "title": new_goal.title,
+            }}
+            return make_response(response_value, 201)
+        else:
+            return make_response({"details":"Invalid data"}, 400)
```

Comment:
Nice use of the `get` method to check for the title key. We could also use the guard clause pattern to check for a failed case + early exit first and remove the indentation for the primary logic:

```suggestion
    elif request.method == "POST":
        request_body = request.get_json()
        
        if "title" not in request_body:
            return make_response({"details":"Invalid data"}, 400)
        
        new_goal = Goal(title=request_body["title"])
        db.session.add(new_goal)
        db.session.commit()
        response_value = {"goal":{
                "id": new_goal.goal_id,
                "title": new_goal.title,
        }}
        return response_value, 201
    
```